news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title:  "The (new) PrestaShop Test Framework"
+subtitle: "And how we’re continually improving the quality of the PrestaShop project"
+date:  2020-02-05 12:00:00
+authors: [SimonGarny]
+icon: icon-checked
+image: /assets/images/theme/meta-logo-build.png
+tags: [QA, tests, community]
+---
+
+For a few months, quality has been a priority for PrestaShop: the QA team and I are proud to announce that we’re making great steps in improving the overall quality of the core project. Automatizing tests is a big part of this process: let’s see what we’re doing, why, and how!
+Note: this article is focused on the Core only, but that’s just a small part of what the QA team is working on. More to come in a future article!
+
+
+## Taking a look back
+
+Historically, the PrestaShop Core continuous integration had a big end-to-end test campaign which purpose was to test the whole application as thoroughly as possible. This campaign was using Selenium as the automation tool, and we were running it in two places:
+
+- The whole campaign was running on every nightly build and for every release
+- A small subset of the campaign was running for every Pull Request on GitHub through Travis CI
+
+At the beginning of 2019, the QA team decided to drop this campaign and start over. In September 2019, we started the new version.
+
+Why? There were mainly 3 reasons:
+
+- Tests are run in headless mode (mandatory on Travis) and sadly, Selenium is kind of picky in headless. There were a lot of problems with it, and as you can imagine it’s way more complicated to debug a failing test when there’s no GUI! Screenshots have been tried, logs, everything possible but ultimately the failing tests had to be disabled as it was impossible to find a good solution.
```

Comment:
It seems like you are talking, not writing 😄 . Try to state it in a more quiet and formal tone 😉.
```suggestion
- Tests were run in headless mode (mandatory on Travis). Unfortunately, Selenium is quite picky when used in headless mode. This created multiple issues when tests would fail although the code was bug free. Alas, debugging a failing test without a GUI is very tedious. We used multiple workarounds (screenshots of failing screens, logging every event, ...) but ultimately we were forced to disable te failing tests as we could not make them run in a reliable manner.
```

Also it would be great to back up these statements about Selenium with external references, to provide reader some external content to understand words he does not know (like headless) and to prove you are not inventing these. See [Pablo's article](https://build.prestashop.com/news/prestashop-in-2019-and-beyond-part-2-pain-points/): he has multiple external references (sometimes it comes from stackoverlow) linked when he explain technical issues.